Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,96 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            bool nonExistantCheckWarning = false;
+
+            try
+            {
+                if (externalBoundary.IIsSelfIntersecting())
+                {
+                    Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
+                    return null;
+                }
+            } catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => x.IIsSelfIntersecting()))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is selfintersecting.");
+                    return null;
+                }
+            }
+            catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => !externalBoundary.IIsContaining(x)))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is not contained by the externalBoundary.");
```

Comment:
Can't we remove the surfaces that are not contained in the external boundary and build the planar surface only from the valid ones? Or is this happening in the `BooleanUnion` already?

Reply:
Guess we could, but we'd need to add checks for the special case of internal boundaries intersecting the external one, and perhaps do a boolean difference there?

And no not solved later, returning null

Reply:
Hmm, this starts to sound a bit philosophical. I would say we either do all:
- `BooleanUnion` on internal boundaries
- `BooleanDifference` internal vs external
- `DistributeOutlines`

And return a clean solution (nurbs can be tricky) OR we do not do any Boolean operations whatsoever. Now I would be a bit confused why did the openings get merged but not cut out of the external boundary?